### DISCUSSION TITLE
Connection declined. Connection can be declined if a previous request is still active

### DISCUSSION THREAD

**Discussion by Ay7ot** [WalletConnect/walletconnect-monorepo#4310] [Developer Support]
This is my first time working with Web3Modal and I'm using react. I'm running into a kind of bug. When Itry to connect from a mobile browser or a browser that doesn't have an extension installed, i get the error Connection declined. Connection can be declined if a previous request is still active. 
My Context provider code looks like this:
`import { createWeb3Modal } from '@web3modal/wagmi/react'
import { defaultWagmiConfig } from '@web3modal/wagmi/react/config'
import { injected, metaMask, safe, walletConnect } from 'wagmi/connectors'
import { WagmiProvider } from 'wagmi'
import { arbitrum, mainnet } from 'wagmi/chains'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

// 0. Setup queryClient
const queryClient = new QueryClient()

// 1. Get projectId at https://cloud.walletconnect.com
const projectId = 'MY_PROJECT_ID'

// 2. Create wagmiConfig
const metadata = {
    name: 'Web3Modal',
    description: 'Web3Modal Example',
    url: 'http://localhost:5173', // origin must match your domain & subdomain
    icons: ['https://avatars.githubusercontent.com/u/37784886']
}

const chains = [mainnet, arbitrum]
const config = defaultWagmiConfig({
    chains, // required
    projectId, // required
    metadata, // required
   
    enableWalletConnect: true, // Optional - true by default
    enableInjected: true, // Optional - true by default
    enableEIP6963: true, // Optional - true by default
    enableCoinbase: true, // Optional - true by default
})

// 3. Create modal
createWeb3Modal({
    wagmiConfig: config,
    projectId,
    enableAnalytics: true // Optional - defaults to your Cloud configuration
})

export function ContextProvider({ children }) {
    return (
        <WagmiProvider config={config}>
            <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
        </WagmiProvider>
    )
}`

I basically followed the doc on installation and I dont know why this is happening.
My button is in my app folder and yes, i wrapped my App in ContextProvider.

**Comment by Michelspyra** [comment]
Fb

**Comment by glitch-txs** [comment]
Hi, please share a minimal reproducible example and steps to reproduce the issue. Thanks!

**Reply by maryannabt** [reply]
I can see this issue reproduced here:
https://lab.web3modal.com/library/wagmi/
With Exodus (browser extension enabled, nothing else connected) - "Connection declined" and extension is not opening.

And also another issue - when Trust Wallet extension is enabled, if I'm trying to connect from "All wallets" screen - it says "Not Detected", BUT if connecting to Trust from "Connect Wallet" screen it opens the extension and works fine. I think in the "Connect Wallet" screen it's coming from detected injected provider. So why in "All wallets" screen the same wallet works different?

Are those knowns issues or I'm missing something?

The issues I described happen on https://lab.web3modal.com/library/wagmi/ but on https://web3modal.com/ example connect button everything works (it says 3.5.6 web3modal version there, so maybe related to that?) 
Where can I find the code of those examples?

Waiting for reply and thanks for your help :)

**Reply by glitch-txs** [reply]
I see, I was able to repro, could you open an issue here please: https://github.com/WalletConnect/web3modal/issues

**Comment by Ay7ot** [comment]
This works in a dApp browser like metamask or trustwallet. If you also put it in a browser where an extension is installed, it also works. 
Only browsers where there isn't an extension or there it's not a dApp browser does not work.

**Reply by maryannabt** [reply]
I have extensions installed and it doesn't work :(
At least not here https://lab.web3modal.com/library/wagmi/

**Reply by eminmammadov** [reply]
In your CORS settings, you need to allow WalletConnect connections. Normally, connections are running in the background for mobile compatibility.

**Reply by MartinLi-coder** [reply]
Where should this CORS configuration be set up?

**Comment by mattrltrent** [comment]
Same issue...

**Comment by imyngvarr** [comment]
The whole problem is in the dashboard https://cloud.reown.com/ you have to verify the project domain through settings and then everything will work. Every application key in reown.com cabinet should be for verified domains. This didn't used to happen when wallectconnect....

**Reply by alhxji** [reply]
@imyngvarr How to do this for expo apps
